mpas_analysis/test/test_timekeeping.py

```diff
@@ -0,0 +1,200 @@
+                                  second=2)
+        self.assertEqual(date1, date2)
+
+        # DDD_hh:mm:ss
+        calendar = 'gregorian_noleap'
+        date1 = stringToRelativedelta('0001_00:00:01',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=0, months=0, days=1, hours=0,
+                                  minutes=0, seconds=1, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        # DDD_hh.mm.ss
+        date1 = stringToRelativedelta('0002_01.00.01',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=0, months=0, days=2, hours=1,
+                                  minutes=0, seconds=1, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        # DDD_SSSSS
+        date1 = stringToRelativedelta('0002_00003',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=0, months=0, days=2, hours=0,
+                                  minutes=0, seconds=3, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        # hh:mm:ss
+        date1 = stringToDatetime('00:00:01')
+        date2 = datetime.datetime(year=1, month=1, day=1, hour=0, minute=0,
+                                  second=1)
+        self.assertEqual(date1, date2)
+
+        # hh.mm.ss
+        calendar = 'gregorian'
+        date1 = stringToRelativedelta('00.00.01',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=0, months=0, days=0, hours=0,
+                                  minutes=0, seconds=1, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        # YYYY-MM-DD
+        date1 = stringToDatetime('0001-01-01')
+        date2 = datetime.datetime(year=1, month=1, day=1, hour=0, minute=0,
+                                  second=0)
+        self.assertEqual(date1, date2)
+
+        # SSSSS
+        date1 = stringToRelativedelta('00005',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=0, months=0, days=0, hours=0,
+                                  minutes=0, seconds=5, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0005-00-00',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1991-01-15'))
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0000-02-00',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1995-11-15'))
+
+        date1 = stringToDatetime('1996-01-15')
+        date2 = stringToRelativedelta('0000-00-20',
+                                      calendar='gregorian')
+        diff = date1-date2
+        self.assertEqual(diff, stringToDatetime('1995-12-26'))
+
```

Comment:
Many of these tests already do test addition and subtraction operators.  I will add some more systematic tests of all operators, though.